addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1816,6 +1816,240 @@ QUnit.test('composer should be focused automatically after clicking on the send
+        return {
+            body: 'coucou',
+            channel_ids: [20],
+            id,
+        };
+    }));
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                if (messageFetchShouldFail) {
+                    throw new Error();
+                }
+            }
+            return this._super(...arguments);
+        },
+    });
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    messageFetchShouldFail = true;
+    await afterNextRender(() => document.querySelector('.o_MessageList_loadMore').click());
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        30,
+        "should still show 30 messages as load more has failed"
+    );
+});
+
+QUnit.test('failure on loading more messages should display error and prompt retry button', async function (assert) {
+    assert.expect(3);
+
+    // first call needs to be successful as it is the initial loading of messages
+    // second call comes from load more and needs to fail in order to show the error alert
+    // any later call should work so that retry button and load more clicks would now work
+    let messageFetchShouldFail = false;
+    this.data['mail.channel'].records.push({
+        channel_type: 'channel',
+        id: 20,
+        is_pinned: true,
+        name: "General",
+    });
+    this.data['mail.message'].records.push(...[...Array(60).keys()].map(id => {
+        return {
+            body: 'coucou',
+            channel_ids: [20],
+            id,
+        };
+    }));
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                if (messageFetchShouldFail) {
+                    throw new Error();
+                }
+            }
+            return this._super(...arguments);
+        },
+    });
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    messageFetchShouldFail = true;
+    await afterNextRender(() => document.querySelector('.o_MessageList_loadMore').click());
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_alertLoadingFailed',
+        "should show loading error message"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_alertLoadingFailedButton',
```

Comment:
It's not clear what's the purpose of this button. Would rename part so that there's at least "retry" in it.